test/test_helper.rb

```diff
@@ -1,45 +1,78 @@
+ENV['RAILS_ENV'] = 'test'
+
 require 'codeclimate-test-reporter'
-CodeClimate::TestReporter.start
+require 'simplecov'
+
+class SimpleCov::Formatter::MergedFormatter
+  def format(result)
+    SimpleCov::Formatter::HTMLFormatter.new.format(result)
+    CodeClimate::TestReporter::Formatter.new.format(result)
+ end
+end
 
-ENV["RAILS_ENV"] = "test"
+if ENV['CODECLIMATE_REPO_TOKEN']
+  CodeClimate::TestReporter.start
+  SimpleCov.formatter = SimpleCov::Formatter::MergedFormatter
+else
+  SimpleCov.start do
+    add_filter '/test/'
+    add_filter '/config/'
+    add_filter '/vendor/'
+
+    add_group 'Controllers', 'app/controllers'
+    add_group 'Models', 'app/models'
+    add_group 'Services', 'app/services'
+    add_group 'Serializers', 'app/serializers'
+    add_group 'Libs', 'lib/'
+  end
+  SimpleCov.formatter = SimpleCov::Formatter::HTMLFormatter
+end
 
 require File.expand_path('../../config/environment', __FILE__)
 require 'rails/test_help'
 require 'minitest/rails'
+require 'webmock_helper'
+require 'mocha/mini_test'
+
+require_dependency 'auth/current_user_provider'
 
 require 'sidekiq/testing'
 Sidekiq::Testing.inline!
 
-require 'webmock/minitest'
-require 'webmock_helper'
-
-require 'mocha/mini_test'
-
 require 'json_expressions/minitest'
 JsonExpressions::Matcher.assume_strict_arrays = false
 JsonExpressions::Matcher.assume_strict_hashes = false
 
+require 'webmock/minitest'
+WebMock.disable_net_connect!(allow: %w(codeclimate.com robohash.org))
 $redis = ConnectionPool.new { MockRedis.new }
 
-class ActiveSupport::TestCase
-  fixtures :all
-  include FactoryGirl::Syntax::Methods
-end
+module ActiveSupport
+  class TestCase
```

Comment:
I was gonna ask why you were nesting with a module but after some research I now realize we should be using this style more.  There's some strange constant-scoping stuff involved in `Foo::Bar` namespacing

>In the case of class Foo::Bar it really should push TWO scopes onto the stack (both Foo and Bar), but it only pushes one, hence we get the "problem"